contributors/design-proposals/storage/local-storage-pv.md

```diff
@@ -761,3 +761,213 @@ The central manager will poll each worker for their discovered volumes and creat
+allow storage providers to report capacity limits per topology domain.
+
+The local volume provisioner needs to be updated in several areas to support dynamic provisioning.
+
+##### Backend Storage Management
+
+Mechanisms to manage local storage sources are varied.
+In addition to current `populator`, `deletor` and `discoverer`, a new component named
+`storageManager` will be introduced in the plugin, to manage the backend storage.
+
+`storageManager` is a general interface, each kind of storage should have its own implementation.
+LVM StorageManager will be introduced in alpha phase, acting as the default StorageManager.
+During start-up, two new environment variables are needed for the provisioner DaemonSet:
+
+* `STORAGE_BACKEND` to determine which StorageManager instance to start
+* `PROVISIONER_NAME` to specify the name of the provisioner, its value matches the Provisioner
+specified in the StorageClass.
+
+**Note:** Currently we already have a `provisionerName` in `RuntimeConfig` of the plugin,
+which is in format of `local-volume-provisioner-${NODE_NAME}-${NODE_ID}`, it is annotated
+in PVs with `AnnProvisionedBy`, to indicate which PVs are up to a certain node.
+To distinguish it from the new env `PROVISIONER_NAME`, we will rename it to `provisionerTag`.
+
+A StorageManager instance is responsible for:
+
+1. Monit available capacity for provisioning, and update the capacity in StorageClass if
+there is a change.
+2. Provide the ability to create volumes, detailed in the volume creation section.
+
+A new field `storageSources` will be added into the configuration ConfigMap, to illustrate available
+storage source for provisioning.
+
+```
+kind: ConfigMap
+metadata:
+  name: local-volume-config
+  namespace: kube-system
+data:
+ storageClassMap: |
+    local-static:
+      hostDir: "/mnt/ssds"
+      mountDir: "/local-ssds"
+    local_dynamic:
+      storageSources:
+        - "volumeGroup1"
+      blockCleanerCommand: "/scripts/lv_release.sh"
+```
+For example for an LVM StorageManager, the list of storageSource could contain volume groups.
+For other StorageManagers, the list could contain other storage sources (e.g. unpartitioned disks).
+Each StorageManager should know the scope and what to do with the input.
+
+##### Reconcile Capacity
+
+The StorageManager is expected to reconcile capacity of StorageClasses in its local
+`storageClassMap` regularly. Particularly, LVM StorageManager behaviors as follows:
+
+1. Walks through classes in its `storageClassMap` regularly
+2. Fetch VGs (volume groups) of each class if `storageSources` is specified
+3. Sum up the `VG size`, and compare it with the capacity in StorageClass API object
+4. Update the capacity in StorageClass object if there is a difference
+
+**Note:** For the alpha phase, thinly-provisioned VGs (overcommit) is not supported,
+We'll consider it in a subsequent phase.
+
+Though a list of VGs can be configured as storage source, it's not recommend to

```

Comment:
Is there a reason why you want to support more than 1 VG per storage class?

Reply:
Actually no need.
As the config is not limited to LVM, a list could be more general.
But as you suggested below, if we have more explicit parameters for LVM, we can certainly have a clearer limit on it.